File: 10714 Ants/solution.c
#include <stdio.h>

int findMin(int a, int b) {
    return a < b ? a : b;
}

int findMax(int a, int b) {
    return a > b ? a : b;
}

int main() {
    int cases;
    scanf("%d", &cases);
    while(cases--) {
        int l, n;
        scanf("%d %d", &l, &n);
        int pos, min = 0, max = 0;
        for(int i = 0; i < n; ++i) {
            scanf("%d", &pos);
            min = findMax(min, findMin(pos, l - pos));
            max = findMax(max, findMax(pos, l - pos));
        }
        printf("%d %d\n", min, max);
    }
    return 0;
}

File: 10714 Ants/solution.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main (String[] args) throws IOException {
		StreamTokenizer st = new StreamTokenizer(System.in);
		st.nextToken();
		int cases = (int)st.nval;
		StringBuilder output = new StringBuilder();
		while(cases-- > 0) {
		    st.nextToken();
    		int l = (int)st.nval;
    		st.nextToken();
    		int n = (int)st.nval, min = 0, max = 0;
    		for(int i = 0; i < n; ++i) {
    		    st.nextToken();
    		    int pos = (int)st.nval;
    		    min = Math.max(min, Math.min(pos, l - pos));
                max = Math.max(max, Math.max(pos, l - pos));
    		}
            output.append(min).append(' ').append(max).append('\n');
		}
		System.out.print(output);
	}
}
